### PR TITLE
torchscript for inception model

### PR DESCRIPTION
**Pull request by dhruv5995**

### CODE REVIEW

File: model/inception3d.py [hunk 1]
```diff
@@ -21,15 +33,15 @@ def compute_pad(self, dim, s):
     def forward(self, x):
         # compute 'same' padding
         (batch, channel, t, h, w) = x.size()
-        #print t,h,w
-        out_t = np.ceil(float(t) / float(self.stride[0]))
-        out_h = np.ceil(float(h) / float(self.stride[1]))
-        out_w = np.ceil(float(w) / float(self.stride[2]))
-        #print out_t, out_h, out_w
+        # print t,h,w
```
Comment (thefirebanks) [inline model/inception3d.py]: If we're not gonna use these, I'd say let's just remove it!

File: model/inception3d.py [hunk 2]
```diff
@@ -107,45 +122,74 @@ def forward(self, x):
...
         x = F.pad(x, pad)
-        #print x.size()        
+        # print x.size()
 
         x = self.conv3d(x)
         if self._use_batch_norm:
             x = self.bn(x)
         if self._activation_fn is not None:
-            x = self._activation_fn(x)
+            x = F.relu(x)
```
Comment (thefirebanks) [inline model/inception3d.py]: Wait, why did we change this? 😮 if `self._activation_fn` is NOT `None`, then shouldn't we use that one?

File: model/inception3d.py [hunk 3]
```diff
@@ -203,9 +234,29 @@ def __init__(self, num_classes=400, spatial_squeeze=True,
...
+            "MaxPool3d_5a_2x2",
+            "Mixed_5b",
+            "Mixed_5c",
+            "Logits",
+            "Predictions",
+        ]
 
         if final_endpoint not in self.VALID_ENDPOINTS:
-            raise ValueError('Unknown final endpoint %s' % final_endpoint)
+            raise ValueError("Unknown final endpoint %s" % final_endpoint)
```
Comment (thefirebanks) [inline model/inception3d.py]: I appreciate the usage of `"` instead of `'` lmao 😝

File: model/inception3d.py [hunk 4]
```diff
@@ -214,140 +265,211 @@ def __init__(self, num_classes=400, spatial_squeeze=True,
...
+            padding=0,
+            activation_fn=None,
+            use_batch_norm=False,
+            use_bias=True,
+            name="logits",
+        )
 
         self.build()
 
+        self.modules_dict = nn.ModuleDict(self._modules)
```
Comment (thefirebanks) [inline model/inception3d.py]: is this required for torchscript?

File: model/inception3d.py [hunk 5]
```diff
@@ -214,140 +265,211 @@ def __init__(self, num_classes=400, spatial_squeeze=True,
...
 
         # backbone, no gradient part
         with torch.no_grad():
             for end_point in freeze_endpoints:
-                if end_point in self.end_points:
-                    x = self._modules[end_point](x) # use _modules to work with dataparallel
+                for k, v in self.end_points.items():
+                    if end_point == k:
+                        x = v(x)
+                # if end_point in self.end_points.keys():
```
Comment (thefirebanks) [inline model/inception3d.py]: Let's delete these commented out lines too I'd say

File: model/inception3d.py [hunk 6]
```diff
@@ -214,140 +265,211 @@ def __init__(self, num_classes=400, spatial_squeeze=True,
...
+            # if end_point in self.end_points.keys():
+            #     x = self.modules_dict[end_point](
+            #         x
+            #     )  # use _modules to work with dataparallel
 
         # head
         x = self.logits(self.dropout(self.avg_pool(x)))
-        if self._spatial_squeeze:
-            logits = x.squeeze(3).squeeze(3)
+        # if self._spatial_squeeze:
```
Comment (thefirebanks) [inline model/inception3d.py]: 😮 Why did we get rid of that `if` statement?

File: sign_recognizer.py [hunk 7]
```diff
@@ -16,27 +16,25 @@
...
 
     def __init__(self, model_path=None, mapping_path=None):
-        # if model_path is None:
-        #     model_path = STAGED_MODEL_DIRNAME / MODEL_FILE
-        # self.model = torch.jit.load(model_path)
-        if mapping_path is None:
-            mapping_path = LABEL_MAPPING_PATH
+        if model_path is None:
+            model_path = STAGED_MODEL_DIRNAME / MODEL_FILE
+            print(model_path)
```
Comment (thefirebanks) [inline sign_recognizer.py]: hehe rogue print statement?

File: sign_recognizer.py [hunk 8]
```diff
@@ -128,42 +129,13 @@ def process_video(video_filepath, start_frame, end_frame):
...
-    pretrained_i3d_model.eval()
-
-    # pretrained_model = PL_resnet50.load_from_checkpoint(checkpoint_path=weight_loc).eval()#.cuda(device=0)
-    return pretrained_i3d_model
-
-
 def main():
     parser = argparse.ArgumentParser(description=__doc__.split("\n")[0])
+    # parser.add_argument(
+    #     "model_path",
```
Comment (thefirebanks) [inline sign_recognizer.py]: I think we should still give the user a potential location of the torchscript model, and if they don't provide one, then we use the default one... I think that's how it's implemented in the FSDL lab too

**Review by thefirebanks — CHANGES_REQUESTED** [review]
Looks 🔥 🔥 🔥 🔥 !!! A couple of minor comments:
- Can we add type hints to all the methods in the `inception3d.py` file please? I think you already started (I'll also start adding them in the rest of the code), it looks great and it's gonna be really helpful for readability/IDE type checks.
- I think there are some merge conflicts, could you please take a look at what those are?
- Then I just left a couple of questions for my own understanding

Other than that, noice!

File: model/inception3d.py (hunk 4 above)
Comment (dhruv5995) [reply model/inception3d.py]: yes

File: model/inception3d.py (hunk 2 above)
Comment (dhruv5995) [reply model/inception3d.py]: torch script doesn't take a function as input

File: model/inception3d.py (hunk 1 above)
Comment (dhruv5995) [reply model/inception3d.py]: done

File: model/inception3d.py (hunk 2 above)
Comment (dhruv5995) [reply model/inception3d.py]: also fixed

File: model/inception3d.py (hunk 6 above)
Comment (dhruv5995) [reply model/inception3d.py]: because it's always true

File: model/inception3d.py (hunk 5 above)
Comment (dhruv5995) [reply model/inception3d.py]: done

File: model/inception3d.py (hunk 6 above)
Comment (dhruv5995) [reply model/inception3d.py]: but fixed this too

**Review by thefirebanks — APPROVED** [review]

**Pull request merged**